libs/nest/config/src/lib/InvalidConfiguration.ts

```diff
@@ -0,0 +1,20 @@
+import { ConfigurationError } from './ConfigurationError'
+
+export class InvalidConfiguration {
+  static allowedMembers = ['then']
+  isConfigured = false
+
+  constructor(errorMessage: string) {
+    return new Proxy(this, {
+      get(target, prop) {
+        if (
+          prop in target ||
+          InvalidConfiguration.allowedMembers.includes(String(prop))
+        ) {
+          return (target as any)[prop]
+        }
+        throw new ConfigurationError(errorMessage)
```

Comment:
since this gets thrown when you access the property, is the configuration error clear enough, that the stack trace wouldn't end up seemingly pretty confusing?

Reply:
See comment below.